bin/zerobufCxx.py

```diff
@@ -166,24 +185,24 @@ def write_declaration(self, file):
         else:
             file.write(NEXTLINE + self.declaration())
 
-    def write_implementation(self, file, classname, extra_op = ""):
+    def write_implementation(self, file, classname=None, extra_op = ""):
         if not self.split_implementation:
             return
 
         impl_function = re.sub(r" final$", "", self.function) # remove ' final' keyword
         impl_function = re.sub(r" = [0-9\.f]+ ", " ", impl_function) # remove default params
 
         if self.ret_val: # '{}'-less body
-            file.write("\n" + self.ret_val + " " + classname +
-                       "::" + impl_function +
-                       "\n{" +
-                       NEXTLINE + self.body)
+            file.write('\n{0} {1}{2}\n{{{3}{4}'.format(self.ret_val,
+                (classname + '::') if classname else '', impl_function,
+                NEXTLINE, self.body))
```

Comment:
I would write self.body in a separate file.write call, not so much for performance, but for readability.
I would even replace NEXTLINE with a function call startNextLine(file) to reduce the number of string concatenations (In some places the look ugly).